src/codegen/statements.rs

```diff
@@ -63,21 +64,109 @@ pub(crate) fn statement(
                     vartab,
                     opt,
                 };
-
                 //If we remove the assignment, we must keep expressions that have side effects
                 init.recurse(&mut params, process_side_effects_expressions);
                 return;
             }
 
             let expr = expression(init, cfg, contract_no, Some(func), ns, vartab, opt);
+
             cfg.add(
                 vartab,
                 Instr::Set {
                     loc: *loc,
                     res: *pos,
-                    expr,
+                    expr: expr.clone(),
                 },
             );
+
+            // Lets check if the declaration is a declaration of a dynamic array
+            if let Expression::AllocDynamicArray(_, _, size, _) = expr.clone() {
+                // If its a variable of uint32, assign it to a temp
+                if let Expression::Variable(_, ref ty, position) = *size {
+                    let num = Expression::Variable(pt::Loc::Codegen, Type::Uint(32), position);
+                    let temp_res = vartab.temp_name("array_size", ty);
+                    cfg.add(
+                        vartab,
+                        Instr::Set {
+                            loc: *loc,
+                            res: temp_res,
+                            expr: num,
+                        },
+                    );
+
+                    cfg.array_lengths_temps.insert(*pos, temp_res);
+                }
+                // If size a uint and bits > 32
+                else if let Expression::Trunc(_, _, ref index) = *size {
+                    if let Expression::Variable(_, _, index) = &**index {
+                        //a number var holding array length
+                        let num = Expression::Variable(pt::Loc::Codegen, Type::Uint(32), *index);
```

Comment:
I don't know what this is trying to achieve. It's already the correct type. Remove.

Reply:
If the size of the allocation is a uint256, this condition handles that. see this:
`uint256[] a = new uint256[](size);`
`ty:uint256[] %a = (alloc uint256[] len (trunc uint32 %value.temp.3))`
where size is a uint256